modal/image.py

```diff
@@ -778,7 +781,28 @@ def my_build_function():
         )
         ```
         """
-        return self.extend(build_function=(raw_function, kwargs))
+        from .functions import _Function
+
+        info = FunctionInfo(raw_f)
+        base_mounts = [_get_client_mount()]
+        for key, mount in info.get_mounts().items():
+            base_mounts.append(mount)
+
+        function = _Function(
+            info,
+            image=self,
+            secret=secret,
+            secrets=secrets,
+            gpu=gpu,
+            base_mounts=base_mounts,
+            mounts=mounts,
+            shared_volumes=shared_volumes,
+            memory=memory,
+            timeout=86400,
```

Comment:
minor diff from before, but the max timeout could be overridden if desired, but now it no longer can. Probably worth allowing for?

Reply:
makes sense!